File: ABC/ABC388/c.py
n = int(input())
a = list(map(int,input().split()))[:n]
count = 0

index = n-2
for i in range(n-1,0,-1):
  while (a[index]*2 > a[i] and index>=0):
    index-=1
  count += index+1;
  
print(count) 

File: ABC/ABC388/c.c
#include <stdio.h>
#include <stdlib.h>

int main(void)
{
  int n;scanf("%d",&n);
  int *a;
  long long int count=0;
  a = (int *)malloc(sizeof(int)*n);
  for(int i=0; i<n; i++){
    scanf("%d",&a[i]);
  }

  int index=n-2;
  for(int i=n-1;i>0;i--){
    while(a[index]*2 > a[i] && index>=0){
      index--;
    }
    count += index+1;
  }
  printf("%lld\n",count);

  free(a);
  return 0;
}

